### PR TITLE
AN-101 Update Travis config

### LINKED ISSUE #96: Update to adhere to coding standards
This plugin should be updated to adhere to WordPress Coding Standards. Much of it does already, it just needs a once-over to e.g. convert indentation to tabs.

### PR DESCRIPTION
**Pull request by kevinfodness**
* Update phpcs configuration to use current ruleset.
* Update Travis configuration to run phpcs, and better align with current best practices.
* Move installation steps from `bin/install-wp-tests.sh` to `.travis.yml`.
* Bump minimum required version to PHP 5.6 due to incompatibility with certain tools (e.g., Composer) required for running builds and tests.
* Fix errors in PHP 5.6 related to Facebook and Heading component parsing.

Precursor to addressing #96

### CODE REVIEW

**Comment by koconder** [comment]
@kevinfodness see comments on #96  in regards to Travis CI support

File: .travis.yml [hunk 1]
```diff
@@ -1,61 +1,137 @@
...
     - php: 7.0
-      env: WP_VERSION=latest
+      env: WP_VERSION=latest PHP_LINT=1
     - php: 7.1
-      env: WP_VERSION=latest
+      env: WP_VERSION=latest PHP_LINT=1
     - php: 7.2
-      env: WP_VERSION=latest
-    - php: 5.6
-      env: WP_VERSION=nightly
```
Comment (mboynes) [inline .travis.yml]: Why remove the nightly version test?

File: .travis.yml [hunk 2]
```diff
@@ -1,61 +1,137 @@
...
-    # - php: 5.6
-    #   env: WP_TRAVISCI=phpcs
+      env: WP_VERSION=latest PHP_LINT=1 WP_PHPCS=1
   fast_finish: true
 
+# Use this to prepare the system to install prerequisites or dependencies.
+# e.g. sudo apt-get update.
```
Comment (mboynes) [inline .travis.yml]: You might want to remove this line, even though it's a comment. Long story, but it might affect the platform on which the build is run. Also, it's not very illustrative since you wouldn't want `sudo` to be used. Frankly. you can kill this whole block since `before_install:` is empty anyway.

File: .travis.yml [hunk 3]
```diff
@@ -1,61 +1,137 @@
...
+        ;;
+      *)
+        echo "Using PHPUnit 4.8"
         composer global require "phpunit/phpunit=4.8.*"
-      fi
-  - composer install
-  - |
-    if [[ ! -z "$WP_VERSION" ]] ; then
-      bash bin/install-wp-tests.sh wordpress_test root '' localhost $WP_VERSION
-    fi
```
Comment (mboynes) [inline .travis.yml]: personally, I'm a fan of the shell script approach. why the change?

File: .travis.yml [hunk 4]
```diff
@@ -1,61 +1,137 @@
...
+  - echo "define( 'JETPACK_DEV_DEBUG', true );" >> wp-tests-config.php
+  - cat wp-tests-config.php
 
+  # Create WordPress database.
+  - mysql -e 'CREATE DATABASE wordpress_test;' -uroot
+
+  # Set up the plugin.
+  - cd $og_dir
+  - mkdir -p "${WP_CORE_DIR}wp-content/plugins/apple-news"
+  - cp -R . "${WP_CORE_DIR}wp-content/plugins/apple-news/"
```
Comment (mboynes) [inline .travis.yml]: why is this block necessary? that is, why is it necessary that the plugin live in that directory?

File: .travis.yml [hunk 5]
```diff
@@ -1,61 +1,137 @@
...
+    esac
+
+  # Set up WordPress configuration.
+  - cd $WP_DEVELOP_DIR
+  - echo $WP_DEVELOP_DIR
+  - cp wp-tests-config-sample.php wp-tests-config.php
+  - sed -i "s/youremptytestdbnamehere/wordpress_test/" wp-tests-config.php
+  - sed -i "s/yourusernamehere/root/" wp-tests-config.php
+  - sed -i "s/yourpasswordhere//" wp-tests-config.php
+  - echo "define( 'JETPACK_DEV_DEBUG', true );" >> wp-tests-config.php
```
Comment (mboynes) [inline .travis.yml]: since jetpack isn't installed, why does this constant need to get set?

File: .travis.yml (hunk 1 above)
Comment (kevinfodness) [reply .travis.yml]: I didn't want the build to fail on anything other than a stable version, but I can look into adding this to `allow_failures` instead.

File: .travis.yml (hunk 2 above)
Comment (kevinfodness) [reply .travis.yml]: 👍

File: .travis.yml (hunk 3 above)
Comment (kevinfodness) [reply .travis.yml]: Better alignment with other Travis configs that we use.

File: .travis.yml (hunk 5 above)
Comment (kevinfodness) [reply .travis.yml]: Good point. I'll remove.

File: .travis.yml (hunk 4 above)
Comment (kevinfodness) [reply .travis.yml]: It's possible that it doesn't need to. I was having some difficulty getting the tests to run properly, and this is one of the steps I took to get everything working. I'll experiment with pulling it out.

Comment (kevinfodness) [reply .travis.yml]: As it turns out, I didn't need this after all. Thanks!

File: .travis.yml (hunk 1 above)
Comment (kevinfodness) [reply .travis.yml]: I added this back, using `master` from the git repo, and added it to `allow_failures`.

File: .travis.yml (hunk 3 above)
Comment (mboynes) [reply .travis.yml]: Most of our themes take this approach, but I _think_ most if not all plugins use the shell script. Themes probably should use it too, though.

Comment (kevinfodness) [reply .travis.yml]: Good call—I can restore this in my next PR.

File: .travis.yml (hunk 1 above)
Comment (mboynes) [reply .travis.yml]: I'd recommend not allowing it to fail. I've never had a false positive from it, but it has warned me of legitimate issues in forthcoming releases. If Travis allows it to fail, nobody will ever notice when it does.

Comment (kevinfodness) [reply .travis.yml]: OK, I'll remove the `allow_failures` block. Thanks!
